lib/manageiq/providers/ibm_cloud/cloud_tools/taggings.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'ibm_cloud_global_tagging'
+
+module ManageIQ
+  module Providers
+    module IbmCloud
+      module CloudTools
+        # Module to hold any tagging SDK specific classes.
+        module GlobalTags
+        end
+
+        # CloudTools wrapper class to enhance the IBM Tagging SDK.
+        class GlobalTag < Base::Sdk
+          private
+
+          # Return a GlobalTagging SDK instance. Requires an internet connection.
+          # @return [IbmCloudGlobalTagging::GlobalTaggingV1]
+          def sdk_client
+            IbmCloudGlobalTagging::GlobalTaggingV1.new(:authenticator => @cloudtools.authenticator)
+          end
+
+          # Create a generator that removes the need for pagination.
+          # @param url [String] Full URL to send to server.
+          # @return [Enumerator] Object to page through results.
+          # @yield [BaseInstance] An instance of the instance class.
+          # @yield [Hash] When no BaseInstance set.
+          def each_resource(call_back, offset: 0, **kwargs)
+            response = request(call_back, :offset => offset, **kwargs)
+            current_count = response.fetch(:offset) + response.fetch(:limit)
+
+            resources = response[:items]
+            resources&.each { |value| yield value } # rubocop:disable Style/ExplicitBlockArgument
+
+            return if resources.empty?
+
+            each_resource(call_back, :offset => current_count, **kwargs)
+          end
```

Comment:
Minor, but this recursion creates a deep stack when a simple loop can do the same thing without the recursion. and thus without hurting the stack depth

```suggestion
          def each_resource(call_back, offset: 0, **kwargs)
            loop do
              response = request(call_back, :offset => offset, **kwargs)

              resources = response[:items]
              break if resources.empty?

              resources&.each { |value| yield value } # rubocop:disable Style/ExplicitBlockArgument

              offset = response.fetch(:offset) + response.fetch(:limit)
            end
          end
```

Reply:
You might even be able to remove the offset parameter, because I think it was only exposed for the recursion itself.

Reply:
The :offset is used in the Tags list_tags API. It tells the server where to start its next page.

Reply:
Reimplemented as loop.